File: tree/230.c
/**
 * 借助中序遍历
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int kthSmallest(struct TreeNode* root, int k) {
    if(root==NULL) return 0;
    struct TreeNode** s = (struct TreeNode**)malloc(sizeof(struct TreeNode*));
    //int* l = (int*)malloc(sizeof(int));
    int length = 0;
    //int size = 0;
    while(true)
    {
        while(root)
        {
            s = (struct TreeNode**)realloc(s, sizeof(struct TreeNode*)*(length+1));
            s[length++] = root;
            root = root->left;
        }
        root = s[--length];
        if(k-- == 1) return root->val;
        root = root->right;
    }
}

File: tree/230.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func kthSmallest(root *TreeNode, k int) int {
    l := make([]*TreeNode, 0)
    for {
        for root != nil {
            l = append(l, root)
            root = root.Left
        }
        if k == 1 {
            return l[len(l) - 1].Val
        }
        root = l[len(l) - 1]
        k--
        l = l[:len(l) - 1]
        root = root.Right
    }
    
}
